addons/sale_management/static/src/fields/sale_order_template_line_field/sale_order_template_line_field.js

```diff
@@ -20,6 +21,46 @@ export class SaleOrderTemplateLineListRenderer extends SectionAndNoteListRendere
         return this.shouldCollapse(this.record, 'is_optional');
     }
 
+    get isCurrentSectionOptional() {
```

Comment:
Unless I'm wrong, you are always checking whether last section is optional, not current one.
What if we are inserting a new line in an existing section (with the dedicated button)? won't the line consider the wrong section to see if it's optional or not?